### PR TITLE
feat: add 'maintaining' status to cluster

### PR DESCRIPTION
**Pull request by ryanwuer**
### Description of your changes

Fixes #

I have:

- [ ] Read and followed Horizon's [contribution process](https://github.com/horizoncd/horizon/CONTRIBUTING.md).
- [ ] Run `make build && make lint` to ensure this PR is ready for review.

### CODE REVIEW

File: core/http/api/v1/cluster/routers.go [hunk 1]
```diff
@@ -23,125 +23,129 @@ import (
...
-func (api *API) RegisterRoute(engine *gin.Engine) {
+// RegisterRoute register routes
+func (a *API) RegisterRoute(engine *gin.Engine) {
 	apiGroup := engine.Group("/apis/core/v1")
 	var routes = route.Routes{
 		{
 			Method:      http.MethodPost,
 			Pattern:     fmt.Sprintf("/applications/:%v/clusters", common.ParamApplicationID),
-			HandlerFunc: api.Create,
+			HandlerFunc: a.Create,
```
Comment (shivamrazorpay) [inline core/http/api/v1/cluster/routers.go:33]: any reason why changed api to a in this file ??
is api keyword already used in this file ?

File: core/common/cluster.go [hunk 2]
```diff
@@ -54,6 +54,8 @@ const (
 	ClusterStatusFreed    = "Freed"
 	ClusterStatusDeleting = "Deleting"
 	ClusterStatusCreating = "Creating"
+	// ClusterStatusMaintaining means no publishing operation is allowed on the cluster currently
+	ClusterStatusMaintaining = "Maintaining"
```
Comment (shivamrazorpay) [inline core/common/cluster.go:58]: why not use `UnderMaintenance` as the status? (Just a Suggestion)

File: core/http/api/v1/cluster/routers.go (hunk 1 above)
Comment (ryanwuer) [reply core/http/api/v1/cluster/routers.go:33]: for consistency with the receiver in api.go

File: core/common/cluster.go (hunk 2 above)
Comment (ryanwuer) [reply core/common/cluster.go:58]: Same reason, consistence with 'Deleting', 'Creating', etc

**Pull request merged**